### PR TITLE
Initial working code

### PR DESCRIPTION
**Pull request by lbdreyer**
This adds a command line tool that allows you to run the tests of all the conda packages that are installed in a conda environment.

An integration test has been included (written with the help of @pelson)

### CODE REVIEW

**Comment by lbdreyer** [comment]
It looks like the integration test currently isn't running in the travis build.

**Comment by pelson** [comment]
> It looks like the integration test currently isn't running in the travis build.

You will need to add `conda_testenv.tests` and `conda_testenv.tests.integration` to the setup.py (here https://github.com/SciTools/conda-testenv/blob/master/setup.py#L13), otherwise, the tests wont be installed. You may also need to include package data (i.e. the recipes).

File: conda_testenv/test_env.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    """
+    installed = install.linked(prefix)
+    sources = []
+    for dist in conda.cli.main_list.get_packages(installed, None):
+        info = install.is_linked(prefix, dist)
+        sources.append(info['link']['source'])
+    return sources
+
+
+def recipe_exists(source):
```
Comment (pelson) [inline conda_testenv/test_env.py]: The name suggests a boolean return value, but that isn't the case.

Stylistically, I wonder if you are better going down the "ask forgiveness" approach:

```
def recipe_directory(distribution_source):
    if exists: return path
    else: raise IOError (or a custom error, or something)
```

and then dealing with that in your code.

File: conda_testenv/test_env.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    """
+    tmpdir = tempfile.mkdtemp()
+    test_files = conda_build_test.create_test_files(m, tmpdir)
+    py_files, pl_files, shell_files = test_files
+    if not (py_files or pl_files or shell_files):
+        return
+    env = os.environ
+    env = prepend_bin_path(env, env_prefix, prepend_prefix=True)
+    conda_build_test.run_tests(m, env, tmpdir, py_files, pl_files, shell_files)
+    shutil.rmtree(tmpdir)
```
Comment (pelson) [inline conda_testenv/test_env.py]: Might make sense to turn this into a `try..finally`.

File: conda_testenv/test_env.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+
+def run_env_tests(env_prefix):
+    """
+    Run all the tests defined in the recipe of all packages in the given
+    environment.
+
+    """
+    config.CONDA_NPY = 00
```
Comment (pelson) [inline conda_testenv/test_env.py]: A comment along the lines of this needing to be defined for MetaData to be created for recipes with a dependency of `numpy x.x`, and that the value itself isn't important, would be helpful here.

File: conda_testenv/tests/integration/test_recipes/b/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+package:
+    name: b
+    version: 1.2
+
+requirements:
+    build:
+        - python
+    run:
+        - python
```
Comment (pelson) [inline conda_testenv/tests/integration/test_recipes/b/meta.yaml:10]: Perhaps add a `numpy x.x` into the build and run deps here, to exercise that CONDA_NPY line.

**Comment by pelson** [comment]
Looks good to me. We both know that it would be much nicer to have unit tests rather than a big old integration tests, but pragmatically, we needed the integration test anyway, and it covers 80% of the testing scope needed. Perhaps when conda & conda-build's test infrastructure is more mature, we can piggy back on that work to improve the unit testing here.

**Comment by lbdreyer** [comment]
@pelson, everything seems fine now :smiley:

**Comment by pelson** [comment]
Very nice! :+1:

**Pull request merged**